### PR TITLE
Upgrade all JavaScript dependencies

### PR DESCRIPTION
**Pull request by nwalters512**
This is me trying to get ahead of Dependabot in a few days.

### CODE REVIEW

File: apps/grader-host/tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "extends": "@prairielearn/tsconfig/tsconfig.app.json",
   "compilerOptions": {
     "rootDir": "./src",
-    "outDir": "./dist"
+    "outDir": "./dist",
```
Comment (jonatanschroeder) [inline apps/grader-host/tsconfig.json:5]: Is this intentional or is prettier messing up JSON files?

File: apps/grader-host/package.json [hunk 2]
```diff
@@ -40,10 +40,10 @@
     "@types/chai": "^4.3.11",
     "@types/dockerode": "^3.3.23",
     "@types/fs-extra": "^11.0.4",
-    "@types/node": "^18.19.4",
+    "@types/node": "^20.11.11",
```
Comment (mwest1066) [inline apps/grader-host/package.json:43]: Yay! 🎉

**Review by mwest1066 — APPROVED** [review]

**Review by jonatanschroeder — APPROVED** [review]

File: apps/grader-host/tsconfig.json (hunk 1 above)
Comment (nwalters512) [reply apps/grader-host/tsconfig.json:5]: `tsconfig.json` is treated as JSONC, so this is not "messing up".

**Comment by nwalters512** [comment]
CI was failing because of https://github.com/bigskysoftware/htmx/issues/2148. I revered to `1.9.9`, which should fix things.

**Pull request merged**